$OpenBSD: patch-mk_bootstrap_mk,v 1.2 2004/04/26 23:48:50 dons Exp $
Make sure we know how to build things. I don't know why this isn't
needed on, e.g. linux

And for some reason amd64/ld doesn't like linking GHC unless I tell it
to use less memory.

--- mk/bootstrap.mk.orig	Wed Feb  4 03:05:04 2004
+++ mk/bootstrap.mk	Sun Apr 18 20:59:29 2004
@@ -8,6 +8,8 @@
 # compile the .hc files, so we have to duplicate that functionality here.
 # The result is unfortunately ugly, but we don't have another choice.
 
+include $(TOP)/mk/suffix.mk
+
 TOP_SAVED := $(TOP)
 TOP:=$(TOP)/ghc
 
@@ -136,6 +138,9 @@
    -u "$(UNDERSCORE)GHCziWeak_runFinalizzerBatch_closure" \
    -u "$(UNDERSCORE)__stginit_Prelude"
 
+ifeq "$(TARGETPLATFORM)" "x86_64-unknown-openbsd"
+HC_BOOT_LD_OPTS+=-Wl,-no-keep-memory
+endif
 
 HC_BOOT_LIBS =
 
